_posts/2019-03-18-openfaas-on-nomad.md

```diff
@@ -0,0 +1,153 @@
+
+As the project name suggests, the OpenFaaS architecture is designed to be open and work with any backend or Docker-supported scheduler. This makes it very appealing to organizations who operate their own proprietary systems and infrastructure. Project founder Alex Ellis and the team have done an amazing job building out the OpenFaaS on Kubernetes and Docker Swarm experience.
+
+However, Nomad was missing from the officially supported offerings. I then stumbled upon a project started by Nic Jackson (HashiCorp Developer Advocate) and his [initial post](https://www.hashicorp.com/blog/functions-as-a-service-with-nomad) about the OpenFaaS Nomad provider. I immediately had to try it out.
+
+I got OpenFaaS and all the required components launched in my own local Nomad + Vagrant + Docker environment in about 15 minutes (including `vagrant up`). One of the first things I tried to get going on top of the core components was the [Kafka connector](https://github.com/openfaas-incubator/kafka-connector), which is in incubation. While attempting to trigger my first function via a Kafka topic, I ran into my first snag. The Nomad provider was a little behind in the feature set of all the official OpenFaaS providers; taking this as a sign that development had stalled, I saw an opportunity to contribute and even take up ownership of the provider.
+
+Upon posting my first issue, I was pleasantly surprised by the quick response from Alex and his invitation to the Slack community. Members from OpenFaaS and HashiCorp were present in Slack and answered my questions promptly. This, combined with my yearning to learn Go provided all the incentive I needed to get involved.
+
+OpenFaaS providers are written in Golang, and it is easy to get started with your own backend using the [faas-provider library](https://github.com/openfaas/faas-provider). I had very little experience in Go prior to this project, so I quickly brushed up on some Golang fundamentals and started on some tasks. How hard could it be?
+
+[![OpenFaaS Nomad and Vault Integration](http://img.youtube.com/vi/DObWEaO4etE/0.jpg)](http://www.youtube.com/watch?v=DObWEaO4etE)
+
+## OpenFaaS Secrets and Vault
+
+Nomad specializes in scheduling, and as a result, faas-nomad requires both Consul and Vault components in addition to Nomad to complete the OpenFaaS feature set. Consul provides the service discovery and powers the function proxy. Vault is used in conjunction with Nomad for providing secrets to the functions as well as the newly released API for managing secrets with the faas-cli.
+
+![openfaas-fargate](/images/openfaas-on-nomad/openfaas_nomad_vault.jpg)
+
+Let's look at the Vault integration more closely:
+
+### Vault faas-nomad startup arguments
+
```

Comment:
might be nice to list the pre-requisites for this to work. I'm guessing you need a nomad installation with vault and consul running before carrying out these steps?

Reply:
Yes, it is assumes those 3 components are provisioned already. Good point.